core/arch/arm/mm/core_mmu_private.h

```diff
@@ -38,6 +38,10 @@ void core_mmu_set_info_table(struct core_mmu_table_info *tbl_info,
 void core_mmu_populate_user_map(struct core_mmu_table_info *dir_info,
 				struct user_ta_ctx *utc);
 
+static inline bool core_mmap_table_end(struct tee_mmap_region *mm)
```

Comment:
I'd prefer `core_mmap_is_end_of_table()`.
Another option is to rename `MEM_AREA_NOTYPE` to `MEM_AREA_END` and test that directly.